scopesim/effects/data_container.py

```diff
@@ -189,4 +197,121 @@ def validate(self, etype):
         etype_colname = utils.real_colname("ETYPE", self.meta.colnames)
         return self.meta[etype_colname] == etype
 
+    @property
+    def plot_filename(self):
+        plot_fname = self.meta["report"]["plot_filename"]
+        if plot_fname is None:
+            plot_fname = self.meta["name"].lower().replace(" ", "_")
+        plot_fformat = self.meta["report"]["plot_file_format"]
+
+        return plot_fname + "." + plot_fformat
+
+
+    @property
+    def table_string(self):
+        if isinstance(self.table, Table):
+            tbl_str = str(self.table).replace("-", "=")
+            hdr = tbl_str.split("\n")[1]
+            tbl_str = hdr + "\n" + tbl_str + "\n" + hdr
+        else:
+            tbl_str = ""
+
+        return tbl_str
+
+    @property
+    def meta_string(self):
+        meta_str = ""
+        max_key_len = max([len(key) for key in self.meta.keys()])
+        for key in self.meta:
+            if key not in ["comments", "changes", "description", "history", "report"]:
+                meta_str += f"    {key.rjust(max_key_len)} : {self.meta[key]}\n"
+
+        return meta_str
+
+    @property
+    def class_description(self):
+        cls_str = ""
+        if hasattr(str, "__doc__"):
+            cls_str = self.__doc__.split("\n")[0]
+
+        return cls_str
+
+    @property
+    def changes_str(self):
+        changes_str = ""
+        if "comments" in self.meta:
+            if "changes" in self.meta["comments"]:
+                for line in self.meta["comments"]["changes"]:
+                    changes_str += line
+
+        return changes_str
+
+    def report(self, filename=None, rst_title_chars="*+"):
+        """
+        For Effect objects, generates a report based on the data and meta-data
+
+        This is to aid in the automation of the documentation process of the
+        instrument packages in the IRDB.
+
+        .. note:: If the Effect cna generate a plot, this will be saved to disc
+
+
+        Parameters
+        ----------
+        filename : str, optional
+            Where to save the RST file
+        rst_title_chars : 2-str, optional
+            Twi unique characters used to denote rst subsection headings.
+            Options: = - ` : ' " ~ ^ _ * + # < >
+
+        Returns
+        -------
+        rst_text : str
+            The full reStructureText string
+
+        """
+        rst_text = f"""
+{self.__repr__()}
```

Comment:
Question: `__repr__` _should_ return a text-representation of the entire object (as in, a computer readable thing), but you want a human readable string, so why not just use `__str__`? (And I think it is more Pythonic to call the magic methods implicitly: `repr(self)`).

Reply:
The only reason, and it's a lazy reason, is because I've definied __repr__ for lots of classes, but not __str__.

Reply:
Good point though. I should stick to the pythonic ways